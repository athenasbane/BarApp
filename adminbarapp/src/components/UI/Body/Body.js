import React from 'react';
import { Container, makeStyles } from '@material-ui/core';
import { Route, Switch } from 'react-router-dom';
import SetProducts from '../../SetProducts/SetProducts';
import SetTables from '../../SetTable/SetTable';
import Login from '../../Login/Login';
import AuthedRoute from '../../Login/AuthedRoute/AuthedRoute';

const useStyles = makeStyles({
  root: {
    width: '100%',
    margin: 0,
    padding: 0,
  },
});

const Body = () => {
  const classes = useStyles();
  return (
    <Container className={classes.root} maxWidth="lg">
      <Switch>
        <Route exact path="/">
          <Login />
        </Route>
        <AuthedRoute path="/products" component={SetProducts} />
        <AuthedRoute path="/tables" component={SetTables} />
        <Route path="/settings">{/* Settings (With Auth Guard) */}</Route>
        <Route path="/downloads">{/* Downloads (With Auth Guard) */}</Route>
        <Route path="/404">{/* 404 */}</Route>
      </Switch>
    </Container>
  );
};

export default Body;
