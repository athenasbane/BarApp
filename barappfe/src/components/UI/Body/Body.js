import React from 'react';
import { Container, makeStyles } from '@material-ui/core';
import Products from '../../Products/Products';

const useStyles = makeStyles({
  root: {
    width: '100%',
    margin: 0,
    padding: 0,
  },
});

export const Body = () => {
  const classes = useStyles();
  return (
    <Container className={classes.root} fixed>
      <Products />
    </Container>
  );
};

export default Body;
