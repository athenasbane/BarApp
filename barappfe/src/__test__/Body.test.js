import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Body } from '../components/UI/Body/Body';
import { Provider } from 'react-redux';

const mockStore = configureStore();
const store = mockStore();

describe('Body Tests', () => {
    test('renders correctly', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <Body />
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
}); 