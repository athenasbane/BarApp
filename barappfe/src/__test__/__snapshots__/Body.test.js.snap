// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body Tests renders correctly 1`] = `ShallowWrapper {}`;
